01-basics/04-MeetupTitle/script.js

```diff
@@ -14,4 +14,71 @@ function fetchMeetupById(meetupId) {
   });
 }
 
-// Требуется создать Vue приложение
+const template = `
+  <div>
+    <label v-for="n of 5" :key="n">
+      <input
+        type="radio"
+        :value="n"
+        v-model="meetupId"
+        :disabled="loading"
+      /> {{ n }}
+    </label>
+    <hr />
+
+    <h3 v-show="!loading">{{ meetupHeader }}</h3>
+    <h4 v-show="loading">loading...</h4>
+  </div>
+`;
+
+const app = createApp({});
+
+const Meetups = defineComponent({
+  data() {
+    return {
+      meetupId: 1,
+      loading: false,
+      meetupData: null,
+    };
+  },
+  methods: {
+    async loadMeetup(id) {
+      // we need to introduce `requestId` here to avoid
+      // race condition and respect the most recent call
+      // seems like an overkill for now
+      if (this.loading) {
+       return Promise.reject('Currently loading');
+      }
+
+      this.loading = true;
+      this.meetupData = null;
+
+      const meetupData = await fetchMeetupById(id);
+
+      this.loading = false;
+      this.meetupData = meetupData;
+    },
+  },
+  computed: {
+    meetupHeader() {
+      return this.meetupData?.title || '';
+    },
+  },
+  created() {
+    this.loadMeetup(this.meetupId);
+  },
+  watch: {
+    meetupId(newId, oldId) {
+      if (newId === oldId) {
+        return;
+      }
+
+      this.loadMeetup(newId);
+    },
```

Comment:
In vue watcher won't execute if the value hasn't been changed. `newValue` can't be equal to `oldValue`.